docs/content/en/docs/Coding Milestones/PoC2023q1/edge-syncer.md

```diff
@@ -0,0 +1,95 @@
+date: 2023-03-22
+weight: 4
+description: >
+---
+
+{{% pageinfo %}}
+Edge syncer is a new syncer working with mailbox workspace in edge-mc. 
+{{% /pageinfo %}}
+
+![edge-syncer drawio](./images/edge-syncer-overview.png)
+
+#### Registering Edge Syncer on a p-cluster
+
+Edge-syncer can be deployed on p-cluster easily by the following steps.
+1. Create SyncTarget and Location
+  - Mailbox controller creates mailbox workspace automatically. 
+  - Get mailbox workspace id
+2. Use command to register edge-syncer
+   ```console
+   kubectl ws <mb-ws name>
+   kcp workload edge-sync <EM Sync Target name> --image <EM Syncer Image> -o edge-syncer.yaml
+   ```
+3. Deploy edge-syncer on p-cluster.
+
+![edge-syncer boot](./images/edge-syncer-boot.png)
+
+#### Deploy workload objects from edge-mc to p-cluster
+
+To deploy resources to p-clusters, create the following in workload management workspace
+- workload objects
+  - Some objects are denatured if needed.
+  - Other objects are as it is
+- APIExport/API Schema corresponding to CRD
+  - TBD: Conversion from CRD to APIExport/APISchema could be automated by using MutatingAdmissionWebhook on workload management workspace.
+- EdgeSyncConfig (This is going to be replaced with EdgePlacement spec) 
+
+![edge-syncer deploy](./images/edge-syncer-deploy.png)
+
+After this, Edge-mc will put the following in the mailbox workspace.
+- Workload objects (both denatured one and not-denatured one)
+- API Binding (to API Export on the workload management workspace)
+- EdgeSyncConfig CR
+
+**TODO**: This is something we should clarify..e.g. which existing controller(s) in edge-mc will cover, or just create a new controller to handle uncovered one. 
+
+#### EdgeSyncConfig
+- The example is [here](https://gist.github.com/yuji-watanabe-jp/dfa9cdf35b06f612818446c2c3eb7969).
+- The CR here is used from edge syncer. 
+- The CR is placed at mb-ws to define
+  - object selector
+  - need of renaturing
+  - need of status upsync for downsyncing
+  - need of delete propagation for downsyncing
+  - Interval of downsyncing/upsyncing
+- The CR is managed by edge-mc
+  - At the initial implementation before edge-mc side controller become ready, we assume EdgeSyncConfig is on workload management workspace (wm-ws), and then which will be copied into mb-ws like other workload objects.
+  - This should be changed to be generated according to EdgePlacement spec. 
+- This CR is a placeholder for defining how edge-syncer behaves, and will be extended/splitted/merged according to further design discussion.
+
+#### Downsyncing
+
+- Edge syncer does downsyncing, which copy workload objects on mailbox workspace to p-cluster
+- If workload objects are deleted on mailbox workspace, the corresponding objects on the p-cluster will be also deleted according to EdgeSyncConfig. 
+- EdgeSyncConfig specifies which objects should be downsynced.
+  - object selector: group, version, kind, name, namespace (for namespaced objects), label, annotation
```

Comment:
Think about verbosity here.  For example, if you want to downsync all the ConfigMap objects in namespace `kyverno` then there is no need to identify them by name as well.  Do you have actual cases where you want to downsync a proper subsset of namespaced objects of a certain kind in a certain namespace?

Reply:
> downsync all the ConfigMap objects in namespace kyverno

```
kind=configmap, namespace=kyverno
```

> downsync only a specific secret `mysecret` in namespace `my-ns`
```
kind=secret, name=mysecret, namespace=my-ns
```

Reply:
Do you have an actual namespace in a workload management workspace from which you want to downsync some of the Secret objects but not all of them?

Reply:
depending on what else is in the namespace. if no other secrets does not exist there, I can send all. if some secrets such as service account token exists, I do not want to send all.